.github/workflows/preview.yml

```diff
@@ -68,8 +68,10 @@ jobs:
       - name: Create diff to old base
         run: |
           diff -Nrwu old/ preview/${{ fromJSON(env.PR_DATA).target_name }}/ | cat > preview/diff.patch
-          pygmentize -o preview/diff.html -l diff -f html -O full preview/diff.patch
-          diffstat -l -p2 preview/diff.patch > diff.txt
+          if [[ -s preview/diff.patch ]] ; then
```

Comment:
Potential performance improvement: only install `diffstat python3-pygments` if there is a diff.